### ISSUE TITLE
This error in Nextcloud

### ISSUE THREAD

**Issue by sruckh** [nextcloud/fulltextsearch_elasticsearch#8]
`Class OCA\FullTextSearch_ElasticSearch\Platform\ElasticSearchPlatform contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (OCA\FullTextSearch\IFullTextSearchPlatform::removeIndex) at /var/www/nextcloud/apps/fulltextsearch_elasticsearch/lib/Platform/ElasticSearchPlatform.php#53`

I have nextcloud 12 running with lighttpd and PHP 7.0.26-2+0 on Raspberry Pi 3 (raspberian).  I have installed (from github), fulltextsearch, files_fulltextsearch, and fulltextsearch_elasticsearch (ran composer install).  I have installed elasticsearch 6.1.1.  I can run simple queries against elasticsearch.  I have run 'sudo -u www-data php -f /var/www/nextcloud/occ fulltextsearch:index' and it appears to complete successfully.

I am getting the above error from my nextcloud logs from the nextcloud administrative GUI.  Although the index has completed (from above), I can not search for words that would be found in a PDF file that is part of my current files that are stored on nextcloud.

I was previously running nextant/SOLR but have wiped that out and gone down the FullTextSearch/ElasticSearch rabbit hole.

Any suggestions on troubleshooting the error?

**Comment by daita** [comment]
if you're using the version from git for fulltextsearch_elasticsearch, please get the [last version from git for fulltextsearch](https://github.com/nextcloud/fulltextsearch/commit/17927a50d8db7cfb7d4e9c571bda60402a1cc086) too.

**Comment by sruckh** [comment]
Thank you. I will try that, but I believe I pulled fulltextsearch, files_fulltextsearch, and fulltextsearch_elasticsearch within minutes of each other.  I will delete apps, and try again.

**Comment by sruckh** [comment]
I did a new pull today.  Disabled/Removed fulltextsearch, files_fulltextsearch, and fulltextsearch_elasticsearch from nextcloud admin GUI.  Copied over new files for the 3 apps to /var/www/nextcloud/apps.  Set the permissions of those directories to match the other apps.  Ran composer in fulltextsearch_elasticsearch directory as the www-data user.  stopped elastic search and deleted all data.  restarted elasticsearch.  Logged into nexcloud and enabled all 3 plugins.  Went to the full text search admin section of nextcloud and set parameters.  Ran occ fulltextsearch:reset and occ fulltextsearch:index.  Logged back into nextcloud and tried a search, and it appears to be working.  Thank You.
